src/test/java/seedu/address/logic/commands/AddCommandIntegrationTest.java

```diff
@@ -42,4 +43,55 @@ public void execute_duplicatePerson_throwsCommandException() {
         assertCommandFailure(new AddCommand(personInList), model, AddCommand.MESSAGE_DUPLICATE_PERSON);
     }
 
+    @Test
+    public void execute_duplicateNameDifferentCase_throwsCommandException() {
+        //eg Cannot add a friend called alex yeoh if Alex Yeoh already exists in Amigos
+        Person personInList = model.getAddressBook().getPersonList().get(0);
+        Person duplicatePerson = new PersonBuilder(personInList).withName(personInList.getName()
+                .fullName.toUpperCase()).build();
+
+        assertCommandFailure(new AddCommand(duplicatePerson), model, AddCommand.MESSAGE_DUPLICATE_PERSON);
+    }
+
+
+    @Test
+    public void execute_duplicateNameSomeOtherFieldsDifferent_throwsCommandException() {
```

Comment:
This is a bit nitpicky, but this test may be fragile in the sense that it relies on your strings being different from the default values in PersonBuilder(), which could change.

Reply:
Hmm oh yeah, agree! Will add a new person and then use PersonBuilder based on the newly added person